ledger/tests/helpers/mod.rs

```diff
@@ -0,0 +1,290 @@
+        // Sample the genesis private key.
+        let private_key = PrivateKey::<CurrentNetwork>::new(rng).unwrap();
+        // Initialize the store.
+        let store = ConsensusStore::<_, LedgerType<_>>::open(StorageMode::new_test(None)).unwrap();
+        // Create a genesis block with a seeded RNG to reproduce the same genesis private keys.
+        let seed: u64 = rng.gen();
+        let genesis_rng = &mut TestRng::from_seed(seed);
+        let genesis_block = VM::from(store).unwrap().genesis_beacon(&private_key, genesis_rng).unwrap();
+
+        // Extract the private keys from the genesis committee by using the same RNG to sample private keys.
+        let genesis_rng = &mut TestRng::from_seed(seed);
+        let private_keys = (0..committee_size).map(|_| PrivateKey::new(genesis_rng).unwrap()).collect();
+
+        Self::from_genesis(private_keys, genesis_block)
+    }
+
+    /// Initialize the builder with the specified commitee and gensis block
+    pub fn from_genesis(private_keys: Vec<PrivateKey<CurrentNetwork>>, genesis_block: Block<CurrentNetwork>) -> Self {
+        // Initialize the ledger with the genesis block.
+        let ledger = Ledger::<CurrentNetwork, LedgerType<CurrentNetwork>>::load(
+            genesis_block.clone(),
+            StorageMode::new_test(None),
+        )
+        .unwrap();
+
+        Self {
+            private_keys,
+            ledger,
+
+            genesis_block,
+            current_height: 1,
+            last_batch: Default::default(),
+            last_committed_batch: Default::default(),
+            last_block_round: 1,
+            round_to_certificates: Default::default(),
+            previous_leader_certificate: Default::default(),
+        }
+    }
+
+    /// Create multiple blocks, with fully-connected DAGs.
+    #[allow(dead_code)]
+    pub fn generate_blocks(&mut self, num_blocks: usize, rng: &mut TestRng) -> Vec<Block<CurrentNetwork>> {
+        self.generate_blocks_with_opts(num_blocks, &BlockOptions::default(), rng)
+    }
+
+    /// Create multiple blocks, with additional parameters
+    pub fn generate_blocks_with_opts(
+        &mut self,
+        num_blocks: usize,
+        options: &BlockOptions,
+        rng: &mut TestRng,
+    ) -> Vec<Block<CurrentNetwork>> {
+        assert!(num_blocks > 0, "Need to build at least one block");
+
+        (0..num_blocks).map(|_| self.generate_block_with_opts(options, rng)).collect()
+    }
+
+    /// Create a new block, with a fully-connected DAG.
+    ///
+    /// This will "fill in " any gaps left in earlier rounds from non participating nodes.
+    pub fn generate_block(&mut self, rng: &mut TestRng) -> Block<CurrentNetwork> {
+        self.generate_block_with_opts(&BlockOptions::default(), rng)
+    }
+
+    /// Same as `generate_block` but with additional options/parameters.
+    pub fn generate_block_with_opts(&mut self, options: &BlockOptions, rng: &mut TestRng) -> Block<CurrentNetwork> {
+        assert!(self.current_height > 0);
```

Comment:
The height of the genesis block is 0.

Reply:
Good catch. I was actually able to remove current_height entirely. It's not used in the code, and that's why the incorrect value did not cause issues.